StyleCop.Analyzers/StyleCop.Analyzers/Settings/ValueOrException`1.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#nullable disable
+
+namespace StyleCop.Analyzers
+{
+    using System;
+
+    /// <summary>
+    /// Stores the result of evaluating a function, either the return value
+    /// or any thrown exception.
+    /// </summary>
+    /// <typeparam name="T">The type of the returned value.</typeparam>
+    internal class ValueOrException<T>
```

Comment:
💡 `System.Lazy<T>` provides this same behavior

Reply:
Did not know that. Excellent!